pkg/descheduler/policyconfig.go

```diff
@@ -45,26 +46,125 @@ func LoadPolicyConfig(policyConfigFile string, client clientset.Interface, regis
 		return nil, fmt.Errorf("failed to read policy config file %q: %+v", policyConfigFile, err)
 	}
 
+	internalPolicy, err := decode(policyConfigFile, policy, client, registry)
+	if err != nil {
+		return nil, err
+	}
+	return internalPolicy, nil
+}
+
+func decode(policyConfigFile string, policy []byte, client clientset.Interface, registry pluginregistry.Registry) (*api.DeschedulerPolicy, error) {
 	versionedPolicy := &v1alpha1.DeschedulerPolicy{}
+	internalPolicy := &api.DeschedulerPolicy{}
+	var err error
 
-	decoder := scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion)
+	decoder := scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion, v1alpha2.SchemeGroupVersion, api.SchemeGroupVersion)
 	if err := runtime.DecodeInto(decoder, policy, versionedPolicy); err != nil {
-		return nil, fmt.Errorf("failed decoding descheduler's policy config %q: %v", policyConfigFile, err)
+		if err.Error() == "converting (v1alpha2.DeschedulerPolicy) to (v1alpha1.DeschedulerPolicy): unknown conversion" {
```

Comment:
What do you mean by returning an error constant?

This error is returned from the runtime library. I can't make it return a constant. And since they are just using fmt.Errorf, error.Is won't work.

https://github.com/kubernetes/apimachinery/blob/6c409361e35e40e38c4056ba0b86647d4244c047/pkg/runtime/codec.go#L69

If I create a constant error and just compare the strings (the only thing that will be equal), I wouldn't be doing anything differently really

Reply:
I agree that checking the error string isn't optimal, but I'm wondering if there is a way to reorganize this so that's not necessary.

It seems weird to me that v1alpha1->internal has to be handled differently than v1alpha2->internal, ie that they both can't be handled by the Decoder. I would expect a function like `Convert_v1alpha1_DeschedulerPolicy_To_api_DeschedulerPolicy()` to call `v1alpha1.V1alpha1ToInternal()` (to handle the necessary manual conversion), and rely on the automated tooling.

If that's not possible, maybe this section can be refactored to not rely on the exact output of the conversion error to determine the branch

Reply:
I responded this here: https://github.com/kubernetes-sigs/descheduler/pull/1006#discussion_r1041244465
v1alpha1 is too different from internal api, and we do too many custom stuff (like how we set thresholdPriorities). 



> If that's not possible, maybe this section can be refactored to not rely on the exact output of the conversion error to determine the branch

I can try that!

Reply:
So, if we can drop the way that we set threshold priorities (that requires a client to be passed around) , Convert_v1alpha1_DeschedulerPolicy_To_api_DeschedulerPolicy calling V1alpha1ToInternal would be possible. Can we do that?

Reply:
Yeah, calling manually-written code is not necessarily the problem, but the arguments of that code, and I guess we don't want to create clients at the the conversion level, or inject clients from the upper level.

Reply:
Ohh, yeah needing a client makes that a lot tougher. My gut tells me that we should look more into how we do that, but it feels like getting close to yak shaving to focus too much on it. So I'm fine with the manual conversion the way it is for now, but would still like to look into the error string check

Reply:
Ack. Let me see what I can do here :)

Reply:
@knelasevero were you able to take another look at this to see if we could avoid the error string literal check?

Reply:
Added a TODO comment that points back to this discussion and also points to my commit in another branch trying to move things around. I could not make it work, but if someone manages to make that work I think that would be the direction to go: 
TODO: https://github.com/kubernetes-sigs/descheduler/pull/1006/commits/eec534af614b2dee0a1406e612770a85918c60eb

COmmit in another branch: https://github.com/knelasevero/descheduler/commit/b25a44c51c0fa31a7831b899e73d83abd12a789a